scripts/deploy-federation.sh

```diff
@@ -215,3 +219,17 @@ CONTEXT="$(kubectl config current-context)"
 for c in ${JOIN_CLUSTERS}; do
   ./bin/kubefed2 join "${c}" --host-cluster-context "${CONTEXT}" --add-to-registry --v=2 ${KF_NS_ARGS}
 done
+
+if [ "`uname`" == 'Darwin' ];then
+
+    # We need to fix cluster ip addr in cluster-registry for mac os.
+    # Assume all context was contained in current kubeconfig.
+    for c in `kubectl config get-contexts -o name`;
+    do
+        k=`kubectl config get-contexts ${c} --no-headers |awk -F "${c}" '{print $2}'|awk '{print $1}'`
+        ip_addr=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ${k}-control-plane)
```

Comment:
i don't think you can assume local docker in deploy-federation.sh

Reply:
For non-local docker , it would be no harm for the patch.  Because it is the cluster ip address anyway. Cluster IP should work in cluster

Reply:
my point is that it isn't always possible to get the address of clusters with `docker inspect`.

Reply:
Please avoid adding docker-specific logic to `deploy-federation.sh`.  I think it is preferable to ensure that cluster urls are correct for the platform in `create-clusters.sh`.